test/src/sdk-harness/test_projects/ec_recover/mod.rs

```diff
@@ -0,0 +1,109 @@
+use fuel_types::Bytes64;
+use fuels::{
+    prelude::*,
+    signers::fuel_crypto::{Message, PublicKey, SecretKey, Signature},
+    tx::Bytes32,
+};
+use rand::{rngs::StdRng, Rng, SeedableRng};
+
+abigen!(
+    EcRecoverContract,
+    "test_projects/ec_recover/out/debug/ec_recover-abi.json"
+);
+
+async fn setup_env() -> Result<
+    (
+        EcRecoverContract,
+        SecretKey,
+        PublicKey,
+        WalletUnlocked,
+        Message,
+        Bytes64,
+    ),
+    Error,
+> {
+    let mut rng = StdRng::seed_from_u64(1000);
+    let msg_bytes: Bytes32 = rng.gen();
+    let private_key = SecretKey::random(&mut rng);
+    let public_key = PublicKey::from(&private_key);
+    let msg = unsafe { Message::from_bytes_unchecked(*msg_bytes) };
+    let sig = Signature::sign(&private_key, &msg);
+    let sig_bytes: Bytes64 = Bytes64::try_from(sig).unwrap();
+    let mut wallet = WalletUnlocked::new_from_private_key(private_key, None);
+
+    let num_assets = 1;
+    let coins_per_asset = 10;
+    let amount_per_coin = 15;
+    let (coins, _asset_ids) = setup_multiple_assets_coins(
+        wallet.address(),
+        num_assets,
+        coins_per_asset,
+        amount_per_coin,
+    );
+    let (provider, _socket_addr) = setup_test_provider(coins.clone(), vec![], None).await;
+    wallet.set_provider(provider);
+
+    let contract_id = Contract::deploy(
+        "test_projects/ec_recover/out/debug/ec_recover.bin",
+        &wallet,
+        TxParameters::default(),
+        StorageConfiguration::default(),
+    )
+    .await?;
+
+    let contract_instance = EcRecoverContract::new(contract_id, wallet.clone());
+
+    Ok((
+        contract_instance,
+        private_key,
+        public_key,
+        wallet,
+        msg,
+        sig_bytes,
+    ))
+}
+
+#[tokio::test]
+async fn can_recover_public_key() {
+    let (contract, _secret, public_key, _wallet, msg, sig_bytes) = setup_env().await.unwrap();
+    let sig_r = &sig_bytes[..32];
+    let sig_v_s = &sig_bytes[32..];
+    let response = contract
+        .methods()
+        .recover_pub_key(
+            Bits256(sig_r.try_into().unwrap()),
+            Bits256(sig_v_s.try_into().unwrap()),
+            Bits256(msg.into()),
+        )
+        .call()
+        .await
+        .unwrap();
+
+    let first = response.value.0;
+    let second = response.value.1;
+    let arrays: [[u8; 32]; 2] = [first.0, second.0];
+    let joined: Vec<u8> = arrays.into_iter().flat_map(|s| s.into_iter()).collect();
+    let joined_array: [u8; 64] = joined.try_into().unwrap();
```

Comment:
Just noting that you should be able to redefine `joined` as an array `joined_array` so that you do not have to create a new name for a variable with a different type. This allows you to reuse the name as a different type.